app/controllers/split_checkout_controller.rb

```diff
@@ -29,7 +29,7 @@ def edit
 
   def update
     if add_voucher
-      return redirect_to checkout_step_path(:payment)
+      return render_voucher_section
```

Comment:
Great work! That's the best solution. I wonder, however, if we should be clear about different types of request. You are assuming a cable ready request here but in theory the form could be submitted as simple HTML request. I've seen two approaches to this:

https://github.com/openfoodfoundation/openfoodnetwork/blob/193ae7936e65dcd7f3dc20bf86bfd45bac37ae9b/app/controllers/admin/enterprises_controller.rb#L50-L55

https://github.com/openfoodfoundation/openfoodnetwork/blob/193ae7936e65dcd7f3dc20bf86bfd45bac37ae9b/app/controllers/concerns/order_stock_check.rb#L32-L38

The latter is clearer to me.

We don't have a convention around this and there is a lot of code in OFN assuming a certain type of request. And I think that we can assume a modern browser with web sockets here. And OFN just doesn't work when Javascript is disabled. But we could do a simple redirect resulting in a reload for HTML in case something breaks web sockets.

Reply:
Did some refactoring and addressed this, I took the second approach. See https://github.com/openfoodfoundation/openfoodnetwork/pull/10587/commits/2bccbf9c4798adb228a76402613845dd3d0597c7

We don't use web sockets here, just usual AJAX request. We use Cable Ready to do the DOM update.